core/node/da_dispatcher/src/da_dispatcher.rs

```diff
@@ -0,0 +1,147 @@
+use std::{future::Future, time::Duration};
+
+use anyhow::Error;
+use chrono::Utc;
+use tokio::sync::watch;
+use zksync_config::DADispatcherConfig;
+use zksync_da_layers::DataAvailabilityInterface;
+use zksync_dal::{Connection, ConnectionPool, Core, CoreDal};
+use zksync_types::L1BatchNumber;
+
+use crate::metrics::METRICS;
+
+#[derive(Debug)]
+pub struct DataAvailabilityDispatcher {
+    client: Box<dyn DataAvailabilityInterface>,
+    pool: ConnectionPool<Core>,
+    config: DADispatcherConfig,
+}
+
+impl DataAvailabilityDispatcher {
+    pub fn new(
+        pool: ConnectionPool<Core>,
+        config: DADispatcherConfig,
+        client: Box<dyn DataAvailabilityInterface>,
+    ) -> Self {
+        Self {
+            pool,
+            config,
+            client,
+        }
+    }
+
+    pub async fn run(self, stop_receiver: watch::Receiver<bool>) -> anyhow::Result<()> {
+        let pool = self.pool.clone();
+        loop {
+            let mut conn = pool.connection_tagged("da_dispatcher").await.unwrap();
+
+            if *stop_receiver.borrow() {
+                tracing::info!("Stop signal received, da_dispatcher is shutting down");
+                break;
+            }
+
+            if let Err(err) = self.dispatch(&mut conn).await {
+                tracing::warn!("dispatch error {err:?}");
+            }
+            if let Err(err) = self.poll_for_inclusion(&mut conn).await {
+                tracing::warn!("poll_for_inclusion error {err:?}");
+            }
+
+            drop(conn);
+            tokio::time::sleep(self.config.polling_interval()).await;
+        }
+        Ok(())
+    }
+
+    async fn dispatch(&self, conn: &mut Connection<'_, Core>) -> anyhow::Result<()> {
+        let batches = conn
+            .blocks_dal()
+            .get_ready_for_da_dispatch_l1_batches(self.config.query_rows_limit() as usize)
+            .await?;
+
+        for batch in batches {
+            let dispatch_latency = METRICS.blob_dispatch_latency.start();
+            let dispatch_response = retry(self.config.max_retries(), || {
+                self.client
+                    .dispatch_blob(batch.l1_batch_number.0, batch.pubdata.clone())
+            })
+            .await
+            .map_err(Error::msg)?;
+            dispatch_latency.observe();
```

Comment:
yeah, I also thought about that, and I'm not sure that we have to track that data for failed requests because they might bring anomalies into the statistics, e.g. taking too little or too much time
and they will be reported as an error in the logs, so we won't miss that something failed